### PR TITLE
Document helpful tips for dev on Apple-silicon/ARM

### LINKED ISSUE #214: Document helpful tips when developing on Apple-silicon/ARM
Documenting "everything found so far when developing on Apple-silicon/ARM":
- gmake
- docker/colima
- integration of TestContainers
- DevContainer for python MLMD wheel being x86 specific

Not including additional "repackaging" of MLMD wheel to be discussed in a different scope.

### PR DESCRIPTION
**Pull request by tarilabs**
Resolves #214 

## Description
bootstrap documentation as discussed.
Follows-up also on:
- https://github.com/opendatahub-io/caikit-tgis-serving/pull/171#issuecomment-1831632015
- https://github.com/opendatahub-io/caikit-tgis-serving/issues/172#issue-2016443950:~:text=An%20initial%20pass%20at%20the%20problem%20has%20been%20discussed%20in%20%23171.

## How Has This Been Tested?
n/a

## Merge criteria:

- [x] The commits and have meaningful messages; the author will squash them [after approval](https://github.com/opendatahub-io/opendatahub-community/blob/main/contributor-cheatsheet.md#:~:text=Usually%20this%20is%20done%20in%20last%20phase%20of%20a%20PR%20revision) or will ask to merge with squash.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by spolti** [comment]
A small change that I do when something fails on apple, is to force podman to use amd64 in the build commands.
podman build --arch=amd64.

works pretty nice.

We could have in the Makefile a if to append args, e.g. when using podman && arm, appends args.

**Comment by tarilabs** [comment]
Ciao @spolti !! 👋 

I've tried Podman, but for some reasons it didn't work well on Mac (at least I've installed all necessary Mac-helpers) specifically when using TestContainers for Go it required code-changes (beyond config changes) unfortunately.
It's a bit sad because I really preferred to use Podman if possible.

I'm not against using it, but at least when developing for this project, so far the best experience was with Colima.

Always happy to hear if anyone can report difference experiences!

**Comment by spolti** [comment]
Thanks for the explanation @tarilabs, how did you install podman?
I didn't install any helper, it works ok (also exposes the podman registry for docker so docker understand that it does have a registry).

All I did was install podman and podman-desktop.

**Comment by tarilabs** [comment]
> Thanks for the explanation @tarilabs, how did you install podman?

Homebrew, as instructed here: https://podman.io/docs/installation#macos:~:text=to%20install%20Podman%3A-,brew%20install%20podman,-Next%2C%20create%20and

The thing is I would like to be sure to leverage Rosetta more than just plain qemu, especially for the DevContainer (per this doc shows). But maybe I misunderstood qemu on Mac...

DevContainer part aside; additionally, I was not able for Testcontainers for Go, to reckon Podman was installed, hence I needed to modify lines such as this one:

https://github.com/opendatahub-io/model-registry/blob/ee6455355f140e3843b56bd17e4185a4e7551b7a/internal/testutils/test_container_utils.go#L18

to specify explicitly podman to be used.
It's not an impossible task, but I needed to make sure each time not-to-commit that change, or to reperform the change locally each time I checkout a PR 😞 

Always happy to hear if anyone get different experience (I'm on Apple-silicon Mac OSX)

**Comment by spolti** [comment]
For that line in specific, maybe not hardcode it?
Instead, use a env, and default to that value if the given env is not set :)

**Comment by tarilabs** [comment]
> Instead, use a env, and default to that value if the given env is not set :)

that is clearly an option, but somehow according to Testcontainers that should not even be needed; unfortunately I didn't have bandwidth to look further into this, given using alternative docker-engine managers just worked

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+==> Caveats
+GNU "make" has been installed as "gmake".
+If you need to use it as "make", you can add a "gnubin" directory
+to your PATH from your bashrc like:
+
+    PATH="/opt/homebrew/opt/make/libexec/gnubin:$PATH"
+...
+```
+
+and now you can substitute `gmake` every time the make command is mentioned in guides (or perform the path management per the caveat).
```
Comment (spolti) [inline CONTRIBUTING.md:55]: isn't better to use just make as stated in the comment? :)

Comment (tarilabs) [reply CONTRIBUTING.md:55]: you mean to adopt the caveat instructions per homebrew install log? :)
that is optional, so I want to keep it same options for our readers (personally I'm just using gmake)

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+## DevContainer
+
+Using a [DevContainer](https://containers.dev) is helpful to develop with the Model Registry Python client, since it needs to wrap MLMD python dependency (which is [x86 specific](https://pypi.org/project/ml-metadata/#files)).
+
+This allows for instance with [VSCode DevContainer extension](https://code.visualstudio.com/docs/devcontainers/containers) to re-open VSCode window "inside" an x86 emulated docker container.
+The experience is very similar to when on GitHub.com you press dot `.` and you get a VSCode "inside something", except it is local to your machine.
+It's not super fast because x86 is emulated via Rosetta, but works "good enough" to complete most tasks without requiring remote connection to a real x86 server.
+
+For non-VSCode user but leveraging LSPs, an SDK is available as well on the upstream: https://containers.dev.
```
Comment (antisaling) [inline CONTRIBUTING.md]: From their docs these editors are supported: https://containers.dev/supporting.html#editors

For my use-case (neovim user) it isn't possible to integrate unless I use a custom extension. Either way, this doesn't have to do with LSPs IIUC.

Comment (tarilabs) [reply CONTRIBUTING.md]: done with  a127c05

**Review by antisaling — APPROVED** [review]

**Pull request merged**